src/parser/ts-patch.js

```diff
@@ -5,9 +5,10 @@ const { replaceRange } = require('./transforms');
 let patchTs, replaceExtensions, syncMtsGtsSourceFiles, typescriptParser, isPatched;
 
 try {
-  const tsPath = require.resolve('@typescript-eslint/parser');
+  const parserPath = require.resolve('@typescript-eslint/parser');
   // eslint-disable-next-line n/no-unpublished-require
-  const ts = require('typescript', { paths: [tsPath] });
```

Comment:
Only `require.resolve` supports `paths`, or options at all. Not a thing for `require`: https://nodejs.org/docs/latest-v20.x/api/modules.html#requireid.

I assume this didn't work as intended, but rather accidentally when `typescript` was resolvable from _this_ package. Which probably explains why this was failing for me with pnpm not allowing this.

Reply:
Yeah, this is how it should be. pass the resolved path to require.